src/Types/Input/InputObjectType.hack

```diff
@@ -11,7 +11,13 @@ abstract class InputObjectType extends NamedType {
 
     <<__Enforceable>>
     abstract const type THackType as shape(...);
-    abstract const keyset<string> FIELD_NAMES;
+    abstract const dict<string, self::TField> FIELDS;
+    const type TField = shape(
```

Comment:
Another minor advantage of the discussed change, we won't need this extra representation for the same thing (`__InputValue` class + this shape)